lib/pages/goal_page.dart

```diff
@@ -3,102 +3,529 @@ import 'package:linear/model/goal.dart';
-          _isErrorFetchingGoals = false;
-        });
-      } else {
-        setState(() {
-          _goals = [];
-          _isLoading = false;
-          _isErrorFetchingGoals = true;
-        });
-      }
-    });
-  }
+ getOpen(List getGoals){
+  
+   setState(() {
+    _goals = _allGoals.toList();
+  });
+   Goal checkGoal = Goal.fromJson(_goals[0]);
+      for(int i = 0; i < _goals.length; i++){
+       checkGoal = Goal.fromJson(_goals[i]);
+       if((checkGoal.isFinished) == true){
+            setState(() {
+             _goals.removeAt(i);
+            i--;
+           });
+       }
+     }
+ }
 
-  @override
-  Widget build(BuildContext context) {
-      return Scaffold(
-        appBar: AppBar(
-          title: const Text("Goals"),
-          automaticallyImplyLeading: false,
-        ),
-        body: Center(
-            child: _isLoading?
-              const CircularProgressIndicator()
-              : _isErrorFetchingGoals?
-                const Text(
-                  "We ran into an error trying to obtain the goals. \nPlease try again later.",
-                  textAlign: TextAlign.center,
-                )
-                : buildGoalsList(),
-              
+ getOverdue(List getGoals){
+  setState(() {
+    _goals = _allGoals.toList();
+  });
+  Goal checkGoal = Goal.fromJson(_goals[0]);
+      for(int i = 0; i < _goals.length; i++){
+       checkGoal = Goal.fromJson(_goals[i]);
+       if((checkGoal.isFinished) == true || (checkGoal.completedCheckIns < checkGoal.checkInGoal)){
+            setState(() {
+             _goals.removeAt(i);
+            i--;
+           });
+       }
+     }
+ }
+
+ getFinished(List getGoals){
+  setState(() {
+    _goals = _allGoals.toList();
+  });
+   Goal checkGoal = Goal.fromJson(_goals[0]);
+      for(int i = 0; i < _goals.length; i++){
+       checkGoal = Goal.fromJson(_goals[i]);
+       if((checkGoal.isFinished) != true){
+            setState(() {
+             _goals.removeAt(i);
+             i--;
+           });
+       }
+     }
+ }
+
+ findIncompleteSearchResults(){
+  String community = "";
+   if(searchInput.text.isEmpty) {
+      return;
+    }else{
+      community = searchInput.text.toString().toLowerCase();
+    }
+  setState(() {
+    _goals = _allGoals.toList();
+    _isSearching = true;
+    isSorting = false;
+  });
+   Goal checkGoal = Goal.fromJson(_goals[0]);
+      for(int i = 0; i < _goals.length; i++){
+       checkGoal = Goal.fromJson(_goals[i]);
+       if(!(checkGoal.communityName).startsWith(community)){
+            setState(() {
+             _goals.removeAt(i);
+            i--;
+           });
+       }
+     }
+     setState(() {
+    _isSearching = false;
+  });
+ }
+
+findSearchResults(){
+  String community = "";
+   if(searchInput.text.isEmpty) {
+      return;
+    }else{
+      community = searchInput.text.toString().toLowerCase();
+    }
+  setState(() {
+    _goals = _allGoals.toList();
+    _isSearching = true;
+    isSorting = false;
```

Comment:
same comment about (don't add all then remove the ones that don't fit the criteria, just add the ones that do).